tests/utils/mock_microgrid.py

```diff
@@ -0,0 +1,207 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Mock microgrid definition."""
+from functools import partial
+from typing import Any, Dict, Set, Union
+from unittest.mock import AsyncMock, MagicMock
+
+from frequenz.channels import Broadcast, Receiver
+from google.protobuf.empty_pb2 import Empty  # pylint: disable=no-name-in-module
+
+from frequenz.sdk.microgrid import Microgrid
+from frequenz.sdk.microgrid._graph import ComponentGraph, _MicrogridComponentGraph
+from frequenz.sdk.microgrid.client import Connection
+from frequenz.sdk.microgrid.component import (
+    BatteryData,
+    Component,
+    ComponentCategory,
+    InverterData,
+)
+
+
+class MockMicrogridClient:
```

Comment:
Could you add a method `initialize` that creates a wrapper `Microgrid` and puts it in `sdk.microgrid._microgrid._MICROGRID`,  so that actors using `get()` will get the mock as well?

Reply:
Good idea!
Done but in a little different way. (last commit)
Because Microgrid has `async def _initialize` method. And I can't force MagicMock to return async from one of its arguments....
I am sure it is possible and I can do it if necessary. But mine solution should be enough.
Please let me know if that works for you :)

Reply:
Oh I didn't realize that you already have a `Microgrid` instance.

I'm starting to think that it is not a good idea at all to put private (or even public) members inside double quotes,  because they will make it impossible for our tooling to detect bugs,  when these mocked names change.

Can you instead just assign `..._MICROGRID = self.mock_microgrid`?  And maybe a `uninitialize` method,  in `__del__` of MockMicrogridClient to set it to None.

Or you can remove the `initialize` method all together.  Now that I know there is a `Microgrid` instance, I can assign it to `_MICROGRID` directly before my tests as well.

Reply:
> Can you instead just assign ..._MICROGRID = self.mock_microgrid inside the new initialize method you added

Done ( in a minute )

> And maybe a uninitialize method, in __del__ of MockMicrogridClient to set it to None.

I think it is not needed. I don't override it. I mock it with pytest mocker. So it won't be problem in the next test.

> Or you can remove the initialize method all together.

Actually I think it is good idea to have it here. User can choose to not call it. And it simplifies test a little.

Reply:
> Can you instead just assign ..._MICROGRID = self.mock_microgrid inside the new initialize method you added

when I said assign,  I meant directly assign,  with an `=` assignment operator,  and not with a mocker. :-)  If you directly assign,  you don't have to put code inside a double quotes,  and our tooling can identify it as such.

Reply:
didn't know that the mocker will make sure the specified variable names exist.  that was my only concern,  so please ignore what I was saying above. :-)